### PR TITLE
Fixes #14975 - Availability Map: Services in Device Groups not displayed correctly

### LINKED ISSUE #14975: Services in Device Groups not displayed correctly in availability map and widget
### The problem

Both Maps -> Availability and the Availability Map Widget for dashboards fail to filter the device group for devices on which these services are running.

Maps -> Availability misses the option to filter on device group for all options/modes (only devices, only services, devices and services). And if a device group is selected, no filtering is applied on the device group for the devices on which these services are running.

The Availability Map Widget uses different code, but has the same issue: services aren't filtered by device group.

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 23.4.0 (2023-04-07T01:52:52+02:00)
DB Schema | 2023_03_14_130653_migrate_empty_user_funcs_to_null (249)
PHP       | 8.2.5
Python    | 3.9.14
Database  | MariaDB 10.5.16-MariaDB
RRDTool   | 1.7.2
SNMP      | 5.9.1
===========================================

[OK]    Composer Version: 2.5.4
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service is enabled
[OK]    Locks are functional
[OK]    No python wrapper pollers found
[OK]    Redis is functional
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
```

### What was the last working version of LibreNMS?

_No response_

### Anything in the logs that might be useful for us?

_No response_

### PR DESCRIPTION
**Pull request by ajsiersema**
Fixes #14975 
Both Maps -> Availability and the Availability Map Widget for dashboards fail to filter the device group for devices on which these services are running.

Maps -> Availability misses the option to filter on device group for all options/modes (only devices, only services, devices and services). And if a device group is selected, no filtering is applied on the device group for the devices on which these services are running.

The Availability Map Widget uses different code, but has the same issue: services aren't filtered by device group.

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by ajsiersema** [comment]
[PHP Static Analysis (8.1) error: 
dbGenPlaceholders() was already used in includes/html/common/availability-map.inc.php, I'd like to change as little code as possible, i.e. not having to refactor all this with Eloquent at this stage

File: includes/html/common/availability-map.inc.php [hunk 1]
```diff
@@ -345,7 +351,7 @@
         </div>
         <div class="page-availability-title-right">';
 
-        if ((Config::get('webui.availability_map_use_device_groups') != 0) && ($mode == 0 || $mode == 2)) {
+        if ((Config::get('webui.availability_map_use_device_groups') != 0) && ($mode == 0 || $mode == 1 || $mode == 2)) {
```
Comment (murrant) [inline includes/html/common/availability-map.inc.php:354]: You can just remove the mode check right?

**Comment by ajsiersema** [comment]
thanks for the re-check assistance, I've commented earlier on the remaining failed lint/PHP static:
https://github.com/librenms/librenms/pull/14976#issuecomment-1511631772

**Comment by murrant** [comment]
An easy way to force a check without actually making changes is to rebase on upstream and push that to your branch.

**Comment by murrant** [comment]
Lint is set to not accept new usages of the deprecated db functions.

**Comment by ajsiersema** [comment]
I'm going to split this up in 2 PR's:
- fix the Availability Map Widget
- find an acceptable new way to implement the Maps -> Availability page, without the deprecated db functions

**Pull request closed without merging**